### PR TITLE
v4l2: Add enumerate_framesizes()

### PR DESCRIPTION
**Pull request by DmitrySamoylov**
Hey!

I've added implementation for framesize enumeration
https://www.kernel.org/doc/html/v4.10/media/uapi/v4l/vidioc-enum-framesizes.html

### CODE REVIEW

**Comment by raymanfx** [comment]
Hi, thanks for this!
Could you please squash the commits into one to keep the history pristine?
I have to say I'm not really happy with adding too many functions to the `capture::Device` struct. Maybe I can come up with a better API here. Will probably think about it for two or three days and then make a decision.

File: examples/framesizes.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+extern crate clap;
+extern crate v4l;
+
+use clap::{App, Arg};
+use v4l::CaptureDevice;
+
+fn main() {
+    let matches = App::new("v4l device")
+        .version("0.2")
+        .author("Christopher N. Hesse <raymanfx@gmail.com>")
```
Comment (raymanfx) [inline examples/framesizes.rs:10]: You are the author of this example ;)

File: src/lib.rs [hunk 2]
```diff
@@ -99,6 +99,9 @@ pub use fourcc::FourCC;
 pub mod format;
 pub use format::FormatDescription;
 
+pub mod framesize;
+pub use framesize::FrameSize;
```
Comment (raymanfx) [inline src/lib.rs:103]: Looks like this should be below `fraction` since 'm' > 'c' alphabetically.

Comment (DmitrySamoylov) [reply src/lib.rs]: Yeah, just overlooked it

**Pull request merged**